packages/web3torrent/src/clients/payment-channel-client.ts

```diff
@@ -258,15 +259,13 @@ export class PaymentChannelClient {
         state.payer === this.mySigningAddress &&
         state.turnNum.mod(2).eq(Index.Payer);
 
-      const currentState = this.channelCache[channelId];
-      if (readyToPay(currentState)) resolve(currentState);
-
-      const unsubscribeListener = this.channelClient.onChannelUpdated(cu => {
-        if (readyToPay(convertToChannelState(cu))) {
-          unsubscribeListener();
-          resolve(convertToChannelState(cu));
-        }
-      });
+      this.channelClient.channelState
+        .pipe(
+          map((result: ChannelResult) => convertToChannelState(result)),
+          takeWhile((latestChannelState: ChannelState) => !readyToPay(latestChannelState), true),
```

Comment:
`takeUntil` might be more clear

Reply:
well... its not the same. takeUntil expects a number of events to complete the subscription. I didn't want to assume that either the first, second or Nth event would be the one that would close the subscription.

Talking with George, we refactored this code to something better

Reply:
Nice, looks good!